aea/helpers/dialogue/base.py

```diff
@@ -656,6 +688,31 @@ def new_self_initiated_dialogue_reference(self) -> Tuple[str, str]:
         """
         return str(self._dialogue_nonce + 1), ""
 
+    def create(self, counterparty, performative, **kwargs,) -> Tuple[Message, Dialogue]:
+        initial_message = self._message_class(
+            dialogue_reference=("", ""),
+            message_id=1,
+            target=0,
+            performative=performative,
+            **kwargs,
+        )
+        initial_message.counterparty = counterparty
+
+        dialogue = self._create_self_initiated(
+            dialogue_opponent_addr=counterparty,
+            role=self.role_from_first_message(initial_message),
+        )
```

Comment:
Why don't we provide role as an argument as well?

Reply:
See above; we create dialogue label before we pass it to this...

Reply:
Done